tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -3,10 +3,14 @@
 import cv2
 import numpy as np
 import scipy.ndimage as ndi
+import torch
+import torchvision
 from skimage import exposure, filters
 from skimage.util import img_as_float
+from torchvision.models._utils import IntermediateLayerGetter
```

Comment:
@John-P is there a way to access intermediate layers without using a private module?

Reply:
You could use forward hooks in pytorch to get features. You may need to know the model architecture to do this though.

Reply:
See https://pytorch.org/docs/stable/generated/torch.nn.modules.module.register_module_forward_hook.html

Reply:
See https://pytorch.org/vision/main/feature_extraction.html

Reply:
Also https://dev.to/jankrepl/forward-hooks-in-pytorch-4eeh

Reply:
@ruqayya Can we use any of the above options?

Reply:
Thanks, John for sharing the links. Now, feature extraction is done using forward hooks.
https://github.com/TissueImageAnalytics/tiatoolbox/blob/3c0a680ca94c281d5cc702fc93369d01d1f617c8/tiatoolbox/tools/registration/wsi_registration.py#L194-L228